### DISCUSSION TITLE
I don't know how to get the enchantment level of an Itemstack since 1.20.5

### DISCUSSION THREAD

**Discussion by pentyum** [FabricMC/fabric-api#3866] [Mod Dev Support] [answered]
### Description

the `EnchantmentHelper.getLevel(enchantment, itemStack)` only accept `RegistryEntry<Enchantment>`，but all enchantments in `Enchantments.class` are `RegistryKey<Enchantment>`. So how can I get the enchantment level?

**Comment by pentyum** [comment, marked as answer]
I have read the implement of `CraftRegistry` in craftbukkit and now have known how to convert `RegistryKey<Enchantment>` to `RegistryEntry<Enchantment>` It should use `server.getRegistryManager().get(RegistryKeys.ENCHANTMENT).getEntry(enchantment_registry_key).get()`.

**Comment by Lithewings** [comment]
Thanks, it helps me a lot, and it also works on 1.21.
When running on client, it works using like that:
world.getRegistryManager().get(RegistryKeys.ENCHANTMENT).getEntry(Enchantments.SILK_TOUCH).get().

**Comment by Jalvaviel** [comment]
Why is it so convoluted now? Is there a reason for this bullcrap?
